torch/csrc/distributed/autograd/utils.cpp

```diff
@@ -42,37 +42,45 @@ DistAutogradContext* addRecvRpcBackward(
     const AutogradMetadata& autogradMetadata,
     std::vector<torch::Tensor>& tensors,
     rpc::worker_id_t fromWorkerId) {
-  TORCH_INTERNAL_ASSERT(
-      torch::autograd::compute_requires_grad(tensors),
-      "Received tensors do not require grad, addRecvRpcBackward should not be called");
   // Initialize autograd context if necessary.
   auto& autogradContainer = DistAutogradContainer::getInstance();
   DistAutogradContext& autogradContext =
       autogradContainer.getOrCreateContext(autogradMetadata.autogradContextId);
 
-  // Attach the tensors as inputs to the autograd function.
-  auto grad_fn = std::make_shared<RecvRpcBackward>(
-      autogradMetadata, autogradContext, fromWorkerId);
-  for (auto& tensor : tensors) {
-    torch::autograd::set_history(tensor, grad_fn);
+  if (!tensors.empty()) {
```

Comment:
Did we catch this issue with the unit tests you added? If not can we add a unit test?

Reply:
Yes, the `to_here()` message is wrapped with autograd, which will trigger this method on the caller. However, we should not insert a recv autograd function for it because that message does not contain any tensor.